internal/resource/http.go

```diff
@@ -263,19 +263,24 @@ func (f *Fetcher) newHttpClient() error {
 	return nil
 }
 
-// getReaderWithHeader performs an HTTP GET on the provided URL with the
+// httpReaderWithHeader performs an HTTP Method on the provided URL with the
 // provided request header and returns the response body Reader, HTTP status
 // code, a cancel function for the result's context, and error (if any). By
 // default, User-Agent is added to the header but this can be overridden.
-func (c HttpClient) getReaderWithHeader(url string, header http.Header) (io.ReadCloser, int, context.CancelFunc, error) {
-	req, err := http.NewRequest("GET", url, nil)
+func (c HttpClient) httpReaderWithHeader(opts FetchOptions, url string) (io.ReadCloser, int, context.CancelFunc, error) {
+	if opts.HTTPVerb == "" {
+		opts.HTTPVerb = "GET"
+	}
+	// `NewRequest` will assign the default value i.e. `GET`
+	// for a given Request.Method if the `verb` is empty.
```

Comment:
As we seem to be setting the default value above this I'm not sure this comment is relevant.